src/utils/hub.js

```diff
@@ -150,6 +150,22 @@ class FileResponse {
     }
 }
 
+/**
+ * Determines whether the given string is a valid Blob URL.
+ * @param {string|URL} string The string to test for validity as an Blob URL.
+ * @returns {boolean} True if the string is a valid Blob URL, false otherwise.
+ */
+function isValidBlobUrl(string) {
```

Comment:
How about abstracting the `isValidHttpURL` function into something like:
```js
function isValidUrl(string, protocols=null) {
  let url;
  
  try {
    url = new URL(string);
  } catch (_) {
    return false;  
  }

  return !protocols || protocols.includes(protocols)
}

function isValidHttpUrl(string) {
    return isValidUrl(string, ["http:", "https:"]);
}
```

and then the other code becomes:
```js
if (env.useFS && !isValidUrl(urlOrPath, ["http:", "https:", "blob:"])) {
```